drivers/gpu/drm/amd/amdgpu/amdgpu_debugfs.c

```diff
@@ -101,9 +101,12 @@ static int amdgpu_debugfs_autodump_open(struct inode *inode, struct file *file)
 
 	file->private_data = adev;
 
+#ifdef __linux__
```

Comment:
That is wrong. You can not skip mutex locking without skipping of unlocking and proving that race appeared is safe.
Our LKPI already has down_write_killable() implementation. At first glance it looks like you need to replace xlock with slock in it to convert it to down_read_killable().
P.S. Really LKPI down_write_killable() is a hack and matches Linux down_write_interruptible() rather than down_write_killable()